SU2_CFD/src/numerics/NEMO/convection/ausm_slau.cpp

```diff
@@ -482,6 +482,88 @@ void CUpwAUSM_NEMO::ComputeInterfaceQuantities(const CConfig* config, su2double*
   for (auto iDim = 0ul; iDim < nDim; iDim++) pressure[iDim] = (P_LP + P_RM) * UnitNormal[iDim];
 }
 
+CUpwAUSMPLUSUP_NEMO::CUpwAUSMPLUSUP_NEMO(unsigned short val_nDim, unsigned short val_nVar,
+                                           unsigned short val_nPrimVar, unsigned short val_nPrimVarGrad,
+                                           const CConfig* config)
+    : CUpwAUSM_SLAU_Base_NEMO(val_nDim, val_nVar, val_nPrimVar, val_nPrimVarGrad, config) {
+
+  Minf = config->GetMach();
+  Kp = 0.25;
+  Ku = 0.75;
+  sigma = 1.0;
+
+  if (Minf < EPS)
+    SU2_MPI::Error("AUSM+Up requires a reference Mach number (\"MACH_NUMBER\") greater than 0.", CURRENT_FUNCTION);
+}
+
+void CUpwAUSMPLUSUP_NEMO::ComputeInterfaceQuantities(const CConfig* config, su2double* pressure,
+                                                    su2double& interface_mach, su2double* interface_soundspeed) {
+
+  /*--- Compute interface speed of sound (aF) ---*/
+
+  const su2double astarL = sqrt(2.0*(Gamma-1.0)/(Gamma+1.0)*Enthalpy_i);
+  const su2double astarR = sqrt(2.0*(Gamma-1.0)/(Gamma+1.0)*Enthalpy_j);
+
+  const su2double ahatL = astarL*astarL/max(astarL, ProjVelocity_i);
+  const su2double ahatR = astarR*astarR/max(astarR,-ProjVelocity_j);
+
+  const su2double A_F = min(ahatL,ahatR);
+
+  /*--- Left and right pressures and Mach numbers ---*/
+
+  su2double mLP, betaLP, mRM, betaRM;
+
+  const su2double mL = ProjVelocity_i/A_F;
+  const su2double mR = ProjVelocity_j/A_F;
+
+  const su2double MFsq = 0.5*(mL*mL+mR*mR);
+  const su2double param1 = max(MFsq, Minf*Minf);
+  const su2double Mrefsq = min(1.0, param1);
+
+  const su2double fa = 2.0*sqrt(Mrefsq)-Mrefsq;
+
+  const su2double alpha = 3.0/16.0*(-4.0+5.0*fa*fa);
+  const su2double beta = 1.0/8.0;
+
+  if (fabs(mL) <= 1.0) {
+    const su2double p1 = 0.25*(mL+1.0)*(mL+1.0);
+    const su2double p2 = (mL*mL-1.0)*(mL*mL-1.0);
+
```

Comment:
I'm recognizing some of this stuff, is it possible to share something with the ideal gas schemes?

Reply:
I actually think all of the these interface Mach functions are dependent on projected velocities, Mach, and pressures.   So yeah all should be able to shared.